lib/cluster/node.js

```diff
@@ -0,0 +1,979 @@
+   *
+   * @param  {string} strategyName
+   * @param  {string} pluginName
+   * @return {void}
+   */
+  onAuthStrategyRemoved (strategyName, pluginName) {
+    this.publisher.sendRemoveAuthStrategy(strategyName, pluginName);
+
+    this.fullState.removeAuthStrategy(strategyName);
+  }
+
+  /**
+   * Triggered when a dump has been requested
+   *
+   * @param  {string} suffix
+   * @return {void}
+   */
+  onDumpRequest (suffix) {
+    this.publisher.sendDumpRequest(suffix);
+  }
+
+  /**
+   * Triggered when security rights have been reset
+   *
+   * @return {void}
+   */
+  onSecurityReset () {
+    this.publisher.send('ResetSecurity', {});
+  }
+
+  /**
+   * Triggered when a document validator has changed
+   *
+   * @return {void}
+   */
+  onValidatorsChanged () {
+    this.publisher.send('RefreshValidators', {});
+  }
+
+  /**
+   * Triggered when a profile has changed
+   *
+   * @param  {string} profileId
+   * @return {void}
+   */
+  onProfileChanged (profileId) {
+    this.publisher.send('InvalidateProfile', { profileId });
+  }
+
+  /**
+   * Triggered when a role has changed
+   *
+   * @param {string} roleId
+   * @return {void}
+   */
+  onRoleChanged (roleId) {
+    this.publisher.send('InvalidateRole', { roleId });
+  }
+
+  /**
+   * Triggered when an index has been added to the index cache
+   *
+   * @param  {storeScopeEnum} scope
+   * @param  {string} index
+   * @return {void}
+   */
+  onIndexAdded (scope, index) {
+    this.publisher.sendAddIndex(scope, index);
+  }
+
+  /**
+   * Triggered when a collection has been added to the index cache
+   *
+   * @param  {storeScopeEnum} scope
+   * @param  {string} index
+   * @param  {string} collection
+   * @return {void}
+   */
+  onCollectionAdded (scope, index, collection) {
+    this.publisher.sendAddCollection(scope, index, collection);
+  }
+
+  /**
+   * Triggered when index have been removed from the index cache
+   *
+   * @param  {storeScopeEnum} scope
+   * @param  {Array.<string>} indexes
+   * @return {void}
+   */
+  onIndexesRemoved (scope, indexes) {
+    this.publisher.sendRemoveIndexes(scope, indexes);
+  }
+
+  /**
+   * Triggered when a collection has been removed from the index cache
+   *
+   * @param  {storeScopeEnum} scope
+   * @param  {string} index
+   * @param  {string} collection
+   * @return {void}
+   */
+  onCollectionRemoved (scope, index, collection) {
+    this.publisher.sendRemoveCollection(scope, index, collection);
+  }
+
+  /**
+   * Triggered when a cluster-wide shutdown has been initiated
+   *
+   * @return {void}
+   */
+  onShutdown () {
+    this.publisher.send('Shutdown', {});
+  }
+
+  /**
+   * Returns the total number of subscribers on the cluster for the provided
+   * room
+   *
+   * @param {string} roomId
+   * @return {void}
+  */
+  async countRealtimeSubscribers (roomId) {
+    return this.fullState.countRealtimeSubscriptions(roomId);
+  }
+}
+
+function checkConfig (config) {
```

Comment:
This should be handled by the config module

Reply:
Funny how I keep telling the same thing when reviewing PRs and I still fail to apply it to myself :sweat: